processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -345,14 +345,23 @@ public static SegmentAnalysis mergeAnalyses(
       mergedId = "merged";
     }
 
+    final Boolean rollup;
+
+    if (arg1.isRollup() != null && arg2.isRollup() != null && !arg1.isRollup() && !arg2.isRollup()) {
+      rollup = Boolean.FALSE;
+    } else {
+      rollup = null;
```

Comment:
I will modify related code to this logic: when merge SegmentAnalysis, false&false=false, true&true=true, true&false=null